src/policies/collections.ts

```diff
@@ -0,0 +1,313 @@
+      hrac,
+      publisher
+    );
+    return new TreasureMap(threshold, destinations, hrac);
+  }
+
+  public static fromBytes(bytes: Uint8Array): TreasureMap {
+    const [thresholdBytes, remainder1] = split(bytes, 1);
+    const [hracBytes, remainder2] = split(remainder1, HRAC.BYTE_LENGTH);
+    const threshold = thresholdBytes.reverse()[0];
+    const nodes = this.bytesToNodes(remainder2);
+    const hrac = new HRAC(hracBytes);
+    return new TreasureMap(threshold, nodes, hrac);
+  }
+
+  private static makeDestinations(
+    ursulas: Ursula[],
+    verifiedKFrags: VerifiedKeyFrag[],
+    hrac: HRAC,
+    publisher: Alice
+  ): KFragDestinations {
+    const destinations: KFragDestinations = {};
+    zip(ursulas, verifiedKFrags).forEach(([ursula, verifiedKFrag]) => {
+      const kFragPayload = AuthorizedKeyFrag.constructByPublisher(
+        hrac,
+        verifiedKFrag,
+        publisher.signer
+      ).toBytes();
+      const ursulaEncryptingKey = PublicKey.fromBytes(
+        fromHexString(ursula.encryptingKey)
+      );
+      destinations[ursula.checksumAddress] = MessageKit.author(
+        ursulaEncryptingKey,
+        kFragPayload,
+        publisher.signer
+      );
+    });
+    return destinations;
+  }
+
+  private static bytesToNodes(bytes: Uint8Array): KFragDestinations {
+    const destinations: KFragDestinations = {};
+    let bytesRemaining = bytes;
+    while (bytesRemaining.length > 0) {
+      const [addressBytes, remainder1] = split(
+        bytesRemaining,
+        ETH_ADDRESS_BYTE_LENGTH
+      );
+      const [messageKit, remainder2] = decodeVariableLengthMessage(remainder1);
+      bytesRemaining = remainder2;
+      const address = toChecksumAddress(addressBytes);
+      destinations[address] = MessageKit.fromBytes(messageKit);
+    }
+    return destinations;
+  }
+
+  public async encrypt(
+    publisher: Alice,
+    bob: Bob,
+    blockchainSigner?: Signer
+  ): Promise<EncryptedTreasureMap> {
+    return EncryptedTreasureMap.constructByPublisher(
+      this,
+      publisher,
+      bob,
+      blockchainSigner
+    );
+  }
+
+  public toBytes(): Uint8Array {
+    return new Uint8Array([
+      // `threshold` must be big-endian
+      ...Uint8Array.from([this.threshold]).reverse(),
+      ...this.hrac.toBytes(),
+      ...TreasureMap.nodesToBytes(this.destinations),
+    ]);
+  }
+
+  private static nodesToBytes(destinations: KFragDestinations): Uint8Array {
+    const kFrag = Object.values(destinations)[0];
+    return Object.entries(destinations)
+      .map(
+        ([ursulaAddress, encryptedKFrag]) =>
+          new Uint8Array([
+            ...toCanonicalAddress(ursulaAddress),
+            ...encodeVariableLengthMessage(encryptedKFrag.toBytes()),
+          ])
+      )
+      .reduce((previous, next) => new Uint8Array([...previous, ...next]));
+  }
+}
+
+export class AuthorizedKeyFrag {
+  public static readonly ENCRYPTED_SIZE = 477; // Hardcoded - depends on Umbral implementation
```

Comment:
Perhaps we need to think about how to keep constants like these up-to-date with what is in `nucypher`?  Even if it is as simple as tests. Maybe the `nucypher-core` work could help with that... @fjarri ?

Reply:
I'm also interested in values defined in `nucypher` using `constant_sorrow`. I would prefer having a single source of truth for parameters shared across different projects (could be `nucypher-core`). One that defines these values "up-front", instead of (for example) calculating them at runtime.

Reply:
In https://github.com/nucypher/nucypher/pull/2802 I'm just wrapping encrypted AKFrag into a `VariableByteString` to get rid of that constant - I'm tired of changing it all the time too.

As for CS values, they're also being removed in 2802.

Reply:
But in any case, if `nucypher-core` takes off, all this logic will be hidden in it.

Reply:
@piotr-roslaniec @fjarri I had a go at defining those constants here… I got distracted by other things, but maybe it would be safe to assume they’d end up something like this https://github.com/nucypher/nucypher/pull/2556

Reply:
Yep, I saw that PR, and that's essentially what I did in 2802. Of course this is all subject to change depending on https://github.com/nucypher/nucypher/issues/2743